docs/tutorials/tabular_prediction/tabular-custom-model.md

```diff
@@ -0,0 +1,374 @@
+# Adding a custom model to AutoGluon
+:label:`sec_tabularcustommodel`
+
+**Tip**: If you are new to AutoGluon, review :ref:`sec_tabularquick` to learn the basics of the AutoGluon API.
+
+This tutorial describes how to add a custom model to AutoGluon that can be trained and ensembled alongside the default models ([default model documentation](../../api/autogluon.tabular.models.html#module-autogluon.tabular.models)).
+
+In this example, we create a custom Random Forest model for use in AutoGluon. All models in AutoGluon inherit from the AbstractModel class ([AbstractModel source code](../../_modules/autogluon/core/models/abstract/abstract_model.html)), and must follow its API to work alongside other models.
+
+Note that while this tutorial provides a basic model implementation, this does not cover many aspects that are used in most implemented models.
+
+To best understand how to implement more advanced functionality, refer to the [source code](../../api/autogluon.tabular.models.html#module-autogluon.tabular.models) of the following models:
+
+- Respecting time limit / early stopping logic
```

Comment:
may add to this list: Example of lazy import of package dependencies

Reply:
added